cluster-state-service/internal/features/wrappers/ec2_wrapper.go

```diff
@@ -52,6 +54,43 @@ func newAWSEC2Session() *session.Session {
 	return sess
 }
 
+type Images []*ec2.Image
+
+func (slice Images) Len() int {
+	return len(slice)
+}
+
+func (slice Images) Less(i int, j int) bool {
+	return *slice[i].CreationDate < *slice[j].CreationDate
+}
+
+func (slice Images) Swap(i int, j int) {
+	slice[i], slice[j] = slice[j], slice[i]
+}
+
+func (EC2Wrapper EC2Wrapper) getLatestECSOptimizedImageID() (string, error) {
+	owner := "amazon"
```

Comment:
What do you mean by "latest"?  Do you mean "highest version number" or "most recently released"?  Those are different things and we do not guarantee that they'll always be the same.